source/documentation/tools/create-a-derived-table/style-guide.md

```diff
@@ -0,0 +1,415 @@
+- 🐍 Schema, table and column names should be in `snake_case`.
+- 🏦 Use names based on the _business_ terminology, rather than the source terminology. For example, if the source database uses `user_id` but the business calls them `customer_id`, use `customer_id` in the model.
+- 🔢 Versions of models should use the suffix `_v1`, `_v2`, etc for consistency (`customers_v1` and `customers_v2`).
+
+### Optional styling
+- 👥 Models should be pluralized, for example, `customers`, `orders`, `products`. Although this is a good best practice, we accept that this may not work with the projects you are working on, so if you cannot keep to it then that is fine.
+- ❌ Do not use abbreviations or aliases. Emphasize readability over brevity. For example, do not use `cust` for `customer` or `o` for `orders`. We accept that in rare cases this may not be possible - if you're not sure, ask for guidance in #ask-data-modelling.
+- 🔙 DBT suggests event dates and times should be past tense. We don't believe this is necessary in our projects, as there are many examples of fields that are well established and changing them would cause confusion. We do however suggest following this for meta data like &mdash; `created`, `updated`, or `deleted`.
+- 🗄️ DBT suggest a consistant ordering of data types in your models, for our use case we do not see this as advantageous as it can be helpful to group fields based on their relevance to eachother, say a flag and the field it is referring to. We therefore advise that a consistent grouping is followed but it does not necessarily need to be based on field type. Where possible ids should be the first fields in a model and we expect the primary key to be **the first** field.
+
+## Example model
+
+Below is an example finance model, following the dbt style of grouping by field type.
+
+```sql
+with
+
+source as (
+
+    select * from {{ source('sop_finance_stg', 'base_hmpps_general_ledger_gl01') }}
+
+),
+
+renamed as (
+
+    select
+
+        ----------  ids
+        id as cost_centre_id, -- primary key
+        a_id as analysis_code_id,
+        objective as cobjective_code_id,
+
+        ---------- strings
+        version
+
+        ---------- numerics
+        cast(debit_amount / 100.0) as float) as debit_amount,
+        cast(credit_amount / 100.0) as float) as credit_amount,
+        cast(total_amount / 100.0) as float) as total_amount,
+        ---------- booleans
+        is_recoverable,
+
+        ---------- dates
+        date(paid_date) as paid_date,
+
+        ---------- timestamps
+        transction_at
+
+    from source
+
+)
+
+select * from renamed
+```
+
+# How we style our SQL
+
+## Basics
+
+- 👻 Use Jinja comments (`{# #}`) for comments that should not be included in the compiled SQL. When dbt compiles your code it will include code using SQL comments like `/* */` and `--`
+- ⏭️ Use trailing commas in lists. e.g.
```

Comment:
maybe we could state that trailing commas are better for readability and maintainability